drivers/SmartThings/jbl/src/lunchbox/rest.lua

```diff
@@ -0,0 +1,403 @@
+local socket = require "cosock.socket"
+local ssl = require "cosock.ssl"
+local log = require "log"
+
+local utils = require "lunchbox.util"
+local Request = require "luncheon.request"
+local Response = require "luncheon.response"
+
+local RestCallStates = {
+  SEND = "Send",
+  RECEIVE = "Receive",
+  RETRY = "Retry",
+  RECONNECT = "Reconnect",
+  COMPLETE = "Complete",
+}
+
+-- build a exponential backoff time value generator
+--
+-- max: the maximum wait interval (not including `rand factor`)
+-- inc: the rate at which to exponentially back off
+-- rand: a randomization range of (-rand, rand) to be added to each interval
+local function backoff_builder(max, inc, rand)
+  local count = 0
+  inc = inc or 1
+  return function()
+    local randval = 0
+    if rand then
+      randval = math.random() * rand * 2 - rand
+    end
+
+    local base = inc * (2 ^ count - 1)
+    count = count + 1
+
+    -- ensure base backoff (not including random factor) is less than max
+    if max then base = math.min(base, max) end
+
+    -- ensure total backoff is >= 0
+    return math.max(base + randval, 0)
+  end
+end
+
+local function connect(client)
+  local default_port = 80
+  local use_ssl = false
+
+  if client.base_url.scheme == "https" then
+    default_port = 443
+    use_ssl = true
+  end
+
+  local port = client.base_url.port or default_port
+
+  local sock, err = client.socket_builder(client.base_url.host, port, use_ssl)
+
+  if sock == nil then
+    client.socket = nil
+    return false, err
+  end
+
+  client.socket = sock
+  return true
+end
+
+local function reconnect(client)
+  client.socket:close()
```

Comment:
This needs a nil check; this was also a bug in the Hue and Sonos versions of Lunchbox that has been fixed there as well.

Reply:
add nil check